### PR TITLE
Issue 3688, add eni to an ecs task

### PR DESCRIPTION
**Pull request by rafcio19**
Resolves https://github.com/spulec/moto/issues/3688

### CODE REVIEW

File: moto/ecs/models.py [hunk 1]
```diff
@@ -318,6 +322,42 @@ def __init__(
...
+                private_ip_address=random_private_ip(),
+                group_ids=net_conf["securityGroups"],
+                description="moto ECS",
+            )
+            eni.status = "in-use"
+            eni.device_index = 0
+
+            self.attachments.append(
+                {
+                    "id": str(uuid.uuid4()),
```
Comment (bblommers) [inline moto/ecs/models.py:349]:
```suggestion
                    "id": str(mock_random.uuid4()),
```
The `mock_random`-object was introduced a week ago or so, and gives us the option to make random identifiers (such as uuid's) deterministic; see http://docs.getmoto.org/en/latest/docs/configuration/recorder/index.html#deterministic-identifiers
Seeding Moto only works if all our identifiers are generated by the same `Random`-object, which is what happens within the `MockRandom`-object

File: moto/ecs/models.py [hunk 2]
```diff
@@ -318,6 +322,42 @@ def __init__(
...
+        if task_definition.network_mode == "awsvpc":
+            if not networking_configuration:
+
+                raise InvalidParameterException(
+                    "Network Configuration must be provided when networkMode 'awsvpc' is specified."
+                )
+
+            self.network_configuration = networking_configuration
+            net_conf = networking_configuration["awsvpcConfiguration"]
+            ec2_backend = ec2_backends["123456789012"]["us-east-1"]
```
Comment (bblommers) [inline moto/ecs/models.py:336]:
```suggestion
            ec2_backend = ec2_backends[self._account_id][self.region_name]
```

Comment (bblommers) [reply moto/ecs/models.py:336]: This should work for all accounts/regions

File: tests/test_ecs/test_ecs_boto3.py [hunk 3]
```diff
@@ -1705,6 +1705,109 @@ def test_run_task():
...
+
+    # Setup
+    client = boto3.client("ecs", region_name="us-east-1")
+    ec2 = boto3.resource("ec2", region_name="us-east-1")
+
+    # ECS setup
+    setup_ecs(client, ec2)
+
+    # Execute
+    with pytest.raises(ClientError):
```
Comment (bblommers) [inline tests/test_ecs/test_ecs_boto3.py:1801]: Would you mind adding some assertions here, to verify that this throws the correct error?
```
with pytest.raises(ClientError) as exc:
    ...
err = exc.value.response["Error"]
err["Code"].equals("?")
err["Message"].equals("?")
```

File: tests/test_ecs/test_ecs_boto3.py [hunk 4]
```diff
@@ -1705,6 +1705,109 @@ def test_run_task():
...
+        UUID(response["tasks"][0]["attachments"][0]["id"])
+    except ValueError:
+        assert False
+    response["tasks"][0]["attachments"][0]["status"].should.equal("ATTACHED")
+    response["tasks"][0]["attachments"][0]["type"].should.equal(
+        "ElasticNetworkInterface"
+    )
+    details = response["tasks"][0]["attachments"][0]["details"]
+
+    subnet_found = False
```
Comment (bblommers) [inline tests/test_ecs/test_ecs_boto3.py:1756]: This entire section may be more readable if we use:
```
assert {"name": "subnetId", "value": setup_resources[0].id} in details
assert {"name": "networkInterfaceId", ..} in details
# etc
```
As far as I can tell, that would test the same thing

**Review by bblommers — CHANGES_REQUESTED** [review]

File: tests/test_ecs/test_ecs_boto3.py (hunk 4 above)
Comment (rafcio19) [reply tests/test_ecs/test_ecs_boto3.py:1756]: You're right, much cleaner and less lines 💪

**Review by bblommers — APPROVED** [review]
LGTM - thanks for fixing this @rafcio19! Much appreciated.

**Pull request merged**